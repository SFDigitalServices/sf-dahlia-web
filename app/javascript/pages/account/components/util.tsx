import { t } from "@bloom-housing/ui-components"
import { ErrorMessages, ErrorMessage } from "./ErrorSummaryBanner"

export const getErrorMessage = (
  errorCode: string,
  errorMessages: ErrorMessages,
  abbreviated?: boolean
) => {
  const error: ErrorMessage = errorMessages[errorCode]
  if (abbreviated && error?.abbreviated) {
    return t(error.abbreviated)
  } else if (error?.default) {
    return t(error.default)
  } else return undefined
}
