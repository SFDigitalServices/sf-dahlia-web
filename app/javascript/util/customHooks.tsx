import { useEffect } from "react"

const tWindow = window as any

const initGoogleTranslate = () => {
  // eslint-disable-next-line no-new
  new tWindow.google.translate.TranslateElement(
    {
      pageLanguage: "en",
      includedLanguages: "en,es,tl,zh-TW",
      layout: tWindow.google.translate.TranslateElement.InlineLayout.HORIZONTAL,
      autoDisplay: false,
      multilanguagePage: true,
    },
    "google_translate_element"
  )
}

export const useTranslate = () => {
  useEffect(() => {
    const script = document.createElement("script")
    script.src = "//translate.google.com/translate_a/element.js?cb=initGoogleTranslate"
    document.body.append(script)
    tWindow.initGoogleTranslate = initGoogleTranslate

    return () => {
      script.remove()
    }
  }, [])
}
