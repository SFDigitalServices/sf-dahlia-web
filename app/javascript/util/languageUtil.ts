import { t, addTranslation } from "@bloom-housing/ui-components"

import { cleanPath, getPathWithoutLeadingSlash } from "./urlUtil"

type PhraseBundle = Record<string, unknown>
export interface LangConfig {
  prefix: LanguagePrefix
  isDefault: boolean
  getLabel: () => string
  load: () => Promise<PhraseBundle>
}

export enum LanguagePrefix {
  English = "en",
  Spanish = "es",
  Chinese = "zh",
  Tagalog = "tl",
}

export const LANGUAGE_CONFIGS: Record<LanguagePrefix, LangConfig> = {
  [LanguagePrefix.English]: {
    prefix: LanguagePrefix.English,
    isDefault: true,
    getLabel: () => t("languages.en"),
    load: async () => await import("../../assets/json/translations/react/en.json"),
  },
  [LanguagePrefix.Spanish]: {
    prefix: LanguagePrefix.Spanish,
    isDefault: false,
    getLabel: () => t("languages.es"),
    load: async () => await import("../../assets/json/translations/react/es.json"),
  },
  [LanguagePrefix.Chinese]: {
    prefix: LanguagePrefix.Chinese,
    isDefault: false,
    getLabel: () => t("languages.zh"),
    load: async () => await import("../../assets/json/translations/react/zh.json"),
  },
  [LanguagePrefix.Tagalog]: {
    prefix: LanguagePrefix.Tagalog,
    isDefault: false,
    getLabel: () => t("languages.tl"),
    load: async () => await import("../../assets/json/translations/react/tl.json"),
  },
}

const loadDefaultTranslation = async (): Promise<PhraseBundle> =>
  await Object.values(LANGUAGE_CONFIGS)
    .find((config) => config.isDefault)
    .load()

/**
 * Load the required translation phrases for the given language prefix.
 *
 * Ex: prefix="es" will load Spanish phrases with English as backup
 *
 * @param prefix is a string, cannot be blank or null.
 */
export const loadTranslations = async (prefix: LanguagePrefix): Promise<void> => {
  addTranslation(await loadDefaultTranslation())

  const config = LANGUAGE_CONFIGS[prefix]

  if (!config.isDefault) {
    addTranslation(await config.load())
  }
}

export const toLanguagePrefix = (routePrefix: string | undefined): LanguagePrefix => {
  switch (routePrefix) {
    case LanguagePrefix.Spanish:
      return LanguagePrefix.Spanish
    case LanguagePrefix.Chinese:
      return LanguagePrefix.Chinese
    case LanguagePrefix.Tagalog:
      return LanguagePrefix.Tagalog
    default:
      return LanguagePrefix.English
  }
}

/**
 * Get the language prefix from the url. Or null if no prefix is on the path
 */
export const getRoutePrefix = (path: string): LanguagePrefix | null => {
  const pathWithoutLeadingSlash = getPathWithoutLeadingSlash(path)
  const langConfig = Object.values(LANGUAGE_CONFIGS).find(
    (languageConfig) =>
      pathWithoutLeadingSlash === `${languageConfig.prefix}` ||
      pathWithoutLeadingSlash.startsWith(`${languageConfig.prefix}/`)
  )

  return langConfig?.prefix
}

/**
 * Get the current path minus the language prefix
 *
 * ex: /es/sign-in -> /sign-in, /sign-in -> /sign-in
 */
export const getPathWithoutLanguagePrefix = (path: string): string => {
  const cleanedPath = cleanPath(path)
  const prefix = getRoutePrefix(cleanedPath)
  if (!prefix) return cleanedPath

  const langConfig = Object.values(LANGUAGE_CONFIGS).find((config) => config.prefix === prefix)
  return cleanPath(`${cleanedPath.replace(langConfig.prefix, "")}`)
}

/**
 * Get the current language prefix, or default to the english prefix if there is no explicit prefix in
 * the path.
 */
export const getCurrentLanguage = (path: string | undefined): LanguagePrefix =>
  getRoutePrefix(path || "") || LanguagePrefix.English
