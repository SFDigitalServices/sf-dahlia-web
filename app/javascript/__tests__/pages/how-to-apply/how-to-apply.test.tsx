import React from "react"
import { cleanup, render, screen, waitFor } from "@testing-library/react"
import { renderAndLoadAsync } from "../../__util__/renderUtils"
import HowToApply, { LeasingAgentBox } from "../../../pages/howToApply/how-to-apply"
import { notYetOpenSaleFcfsListing } from "../../data/RailsSaleListing/listing-sale-fcfs-not-yet-open"
import { fcfsSaleListing } from "../../data/RailsSaleListing/listing-sale-fcfs"
import { openFcfsSaleListing } from "../../data/RailsSaleListing/listing-sale-fcfs-open"
import { localizedFormat, formatTimeOfDay } from "../../../util/languageUtil"

const axios = require("axios")

jest.mock("axios")

jest.mock("react-helmet-async", () => {
  return {
    HelmetProvider: ({ children }: { children: React.ReactNode }) => children, // Mock HelmetProvider
    Helmet: ({ children }: { children: React.ReactNode }) => children, // Mock Helmet component
  }
})

describe("<HowToApply />", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("renders HowToApply component", async () => {
    axios.get.mockResolvedValue({ data: { listing: fcfsSaleListing } })
    const { asFragment } = await renderAndLoadAsync(<HowToApply assetPaths={{}} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("shows 'SUBMIT APPLICATION' button if URL is present", async () => {
    process.env.FCFS_FORMASSEMBLY_URL_EN = "https://www.test.com"
    axios.get.mockResolvedValue({ data: { listing: openFcfsSaleListing } })
    const { queryByText } = await renderAndLoadAsync(<HowToApply assetPaths={{}} />)
    expect(queryByText("Submit application")).not.toBeNull()
  })

  it("opens correct url for a locale", async () => {
    process.env.FCFS_FORMASSEMBLY_URL_EN = "https://www.test.com"
    process.env.FCFS_FORMASSEMBLY_URL_ES = "https://www.test-es.com"
    // set client to ES locale
    const originalLocation = { ...window.location }
    jest.spyOn(window, "location", "get").mockImplementation(() => ({
      ...originalLocation,
      pathname: "/es/test",
    }))
    axios.get.mockResolvedValue({ data: { listing: openFcfsSaleListing } })
    global.open = jest.fn()

    await renderAndLoadAsync(<HowToApply assetPaths={{}} />)
    screen.getByText("Submit application").click()
    await waitFor(() => {
      expect(global.open).toHaveBeenCalledWith(
        `${process.env.FCFS_FORMASSEMBLY_URL_ES}?ListingID=a0W0P00000GlKfBUAV`,
        "_blank"
      )
    })
    jest.restoreAllMocks()
    Object.defineProperty(window, "location", {
      value: originalLocation,
      writable: true,
    })
  })

  it("does not show 'SUBMIT APPLICATION' button if URL is missing", async () => {
    process.env.FCFS_FORMASSEMBLY_URL_EN = undefined
    axios.get.mockResolvedValue({ data: { listing: openFcfsSaleListing } })
    const { queryByText } = await renderAndLoadAsync(<HowToApply assetPaths={{}} />)
    expect(queryByText("Submit application")).toBeNull()
  })

  it("shows the correct header text", async () => {
    const listingData = { data: { listing: fcfsSaleListing } }
    axios.get.mockResolvedValue(listingData)
    const { getAllByText } = await renderAndLoadAsync(<HowToApply assetPaths={{}} />)
    expect(getAllByText(`Apply to ${listingData.data.listing.Name}`)).not.toBeNull()
  })

  it("renders not-yet-open components", () => {
    axios.get.mockResolvedValue({ data: { listing: notYetOpenSaleFcfsListing } })
    const { findByText } = render(<HowToApply assetPaths={{}} />)
    const datetime = notYetOpenSaleFcfsListing.Application_Start_Date_Time || ""
    const date = localizedFormat(datetime, "LL")
    const time = formatTimeOfDay(datetime)
    expect(findByText(`Applications open: ${date} at ${time} Pacific Time`)).toBeDefined()
    expect(
      findByText(`Applications open ${date} at ${time} Pacific Time. Check back here to apply.`)
    ).toBeDefined()
  })

  it("redirects to home page if listing not found", () => {
    axios.get.mockResolvedValue({ data: { listing: null } })
    render(<HowToApply assetPaths={{}} />)
    expect(window.location.pathname).toBe("/")
  })

  it("shows the correct leasing agent box for the listing", async () => {
    const { asFragment } = await renderAndLoadAsync(<LeasingAgentBox listing={fcfsSaleListing} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
