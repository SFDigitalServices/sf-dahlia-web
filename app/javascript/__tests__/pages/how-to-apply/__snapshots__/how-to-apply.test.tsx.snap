// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowToApply /> renders HowToApply component 1`] = `
<DocumentFragment>
  <div
    class="notranslate site-wrapper"
  >
    <div
      class="site-content"
    >
      <title
        class="notranslate"
      >
        Apply to TEST Sale Listing (do not modify) - Homeownership Acres
      </title>
      <meta
        charset="utf-8"
      />
      <meta
        content="IE=edge"
        http-equiv="X-UA-Compatible"
      />
      <meta
        content="width=device-width, initial-scale=1"
        name="viewport"
      />
      <meta
        content="M1zcm6GGM6sHSF_jvkq254DbYAj94JYbFC7ArZDAXlg"
        name="google-site-verification"
      />
      <meta
        content="http://localhost/"
        property="og:url"
      />
      <meta
        content="website"
        property="og:type"
      />
      <meta
        content="Apply to TEST Sale Listing (do not modify) - Homeownership Acres"
        property="og:title"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        property="og:description"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        name="description"
      />
      <meta
        property="og:image"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        property="og:site_name"
      />
      <meta
        content="summary_large_image"
        name="twitter:card"
      />
      <header
        class="site-header"
      >
        <a
          class="site-header__skip-link"
          href="#main-content"
        >
          Skip to main content
        </a>
        <div
          class="language-bar"
        >
          <div
            class="language-bar__inner"
          >
            <nav
              aria-label="Choose your language"
              class="language-nav"
            >
              <ul
                class="language-nav__list"
              >
                <li>
                  <button
                    class="language-nav__list-button is-active"
                    type="button"
                  >
                    English
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    Español
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    中文
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    Filipino
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        <div
          class="site-header__notice "
        >
          <div
            class="site-header__notice-text"
          >
            <span>
              We'd love to get 
              <a
                href="https://airtable.com/shrw64DubWTQfRkdo"
                target="_blank"
              >
                your feedback.
              </a>
            </span>
          </div>
        </div>
        <nav
          aria-label="main navigation"
          class="site-header__container"
          role="navigation"
        >
          <div
            class="site-header__base  site-header__base-wrapped"
          >
            <div
              class="site-header__logo-container site-header__logo-width-med"
              id="site-header-logo"
            >
              <a
                aria-label="DAHLIA San Francisco Housing Portal"
                class="site-header__logo translate site-header__custom-width"
                href="/"
                target="_self"
              >
                <div
                  class="site-header__logo-content site-header__image-only-container"
                >
                  <img
                    alt="Site logo"
                    class="site-header__logo-image site-header__image-only"
                  />
                </div>
              </a>
            </div>
            <div
              class="site-header__navbar-menu site-header__navbar-wrapped"
              id="site-header-links"
            >
              <a
                class="site-header__link  "
                data-testid="Rent-0"
                href="/listings/for-rent"
                target="_self"
              >
                <div>
                  Rent
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Buy-1"
                href="/listings/for-sale"
                target="_self"
              >
                <div>
                  Buy
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="My Favorites-2"
                href="/favorites"
                target="_self"
              >
                <div>
                  My Favorites
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Get Assistance-3"
                href="/get-assistance"
                target="_self"
              >
                <div>
                  Get Assistance
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Sign In-4"
                href="/sign-in"
                target="_self"
              >
                <div>
                  Sign In
                </div>
              </a>
            </div>
          </div>
        </nav>
      </header>
      <main
        class="md:overflow-x-hidden"
        data-testid="main-content-test-id"
        id="main-content"
      >
        <header
          class="page-header is-inverse"
        >
          <hgroup
            class="page-header__group"
          >
            <h1
              class="page-header__title"
              data-testid="page-header"
            >
              Apply to TEST Sale Listing (do not modify) - Homeownership Acres
            </h1>
            <p
              class="page-header__lead"
            >
               Apply to buy a home without entering a lottery. Your application will be considered in the order it comes in — first come, first served.
            </p>
          </hgroup>
        </header>
        <article
          class="flex flex-wrap flex-col md:flex-row relative max-w-5xl m-auto w-full"
        >
          <div
            class="w-full md:w-2/3"
            data-testid="info-main-content"
          >
            <section
              class="flex"
            >
              <article
                class="markdown max-w-5xl m-auto"
              >
                <div
                  class="pt-4 md:py-0 max-w-3xl"
                >
                  <div
                    class="my-6 px-6 md:my-12 md:px-0 md:mr-24"
                  >
                    <div
                      class="seeds-common-message is-fullwidth seeds-message justify-start leading-5"
                      data-variant="primary"
                    >
                      <span
                        class="ui-icon ui-medium"
                      >
                        <svg
                          fill="currentColor"
                          viewBox="0 0 32 32"
                        >
                          <path
                            d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                          />
                          <path
                            d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                          />
                        </svg>
                      </span>
                      <span
                        data-part="content"
                      >
                        <b>
                          We are not accepting applications yet.
                        </b>
                        <br />
                        Applications open: January 1, 2050 at 1:00 AM Pacific Time
                      </span>
                    </div>
                    <div
                      class="py-10"
                    >
                      <h2
                        class="text-2xl font-alt-serif"
                      >
                        How long it takes
                      </h2>
                      <h3
                        class="text-heading-xl font-semibold font-alt-sans pt-6 pb-4"
                      >
                        Preparing your application
                      </h3>
                      It depends on how easy it is for you to gather all your information and documents. It may take a few hours, or you may need to do it over several days. If you applied before, it might take you less time.
                      <h3
                        class="text-heading-xl font-semibold font-alt-sans pt-6 pb-4"
                      >
                        Submitting your application
                      </h3>
                      30 minutes or less
                    </div>
                    <div
                      class="pb-10"
                    >
                      <h2
                        class="text-2xl font-alt-serif"
                      >
                        Before you start
                      </h2>
                      <h3
                        class="text-heading-xl font-semibold font-alt-sans pt-6 pb-4"
                      >
                        Make sure you are eligible
                      </h3>
                      <div>
                        To apply for this unit you must:
                      </div>
                      <ul
                        class="mb-0 pt-2"
                      >
                        <li>
                          <span>
                            Not have owned residential property in the past 3 years
                          </span>
                        </li>
                        <li>
                          <span>
                            Meet our 
                            <a
                              class="underline"
                              href="https://sf.gov/determine-if-you-can-buy-affordable-housing-program"
                              target="_blank"
                            >
                              income requirements
                            </a>
                          </span>
                        </li>
                        <li>
                          <span>
                            Have completed 
                            <a
                              class="underline"
                              href="https://sf.gov/sign-complete-homebuyer-education"
                              target="_blank"
                            >
                              homebuyer education
                            </a>
                          </span>
                        </li>
                        <li>
                          <span>
                            Be pre-approved for a mortgage by a 
                            <a
                              class="underline"
                              href="https://sf.gov/reports/october-2023/find-lender-below-market-rate-program"
                              target="_blank"
                            >
                              lender on our list
                            </a>
                          </span>
                        </li>
                        <li>
                          <span>
                            Have enough in savings for closing costs and downpayment
                          </span>
                        </li>
                      </ul>
                      <h3
                        class="text-heading-xl font-semibold font-alt-sans pt-6 pb-4"
                      >
                        See the unit in person
                      </h3>
                      This is not required, but we encourage you to make sure the unit is right for you before applying.
                      <div
                        class="how-to-apply-dropdown pt-2"
                      >
                        <button
                          aria-expanded="false"
                          class="button is-unstyled m-0 no-underline has-toggle"
                          type="button"
                        >
                          Make an appointment
                        </button>
                      </div>
                    </div>
                    <h2
                      class="text-2xl font-alt-serif"
                    >
                      Apply to
                    </h2>
                    <div
                      class="pt-4"
                      id="HowToApplySection"
                    >
                      <ol
                        class="process-list"
                      >
                        <li>
                          <h3
                            class="text-heading-xl font-semibold font-alt-sans pb-4"
                          >
                            Fill out PDF application
                          </h3>
                          <div
                            class="text-base"
                          >
                            <div
                              class="text-base pb-2"
                            >
                              Download and fill out the application on your computer or print and fill out.
                            </div>
                            <div
                              class="text-base"
                            >
                              If you print the application, you will need to be able to scan it back into your computer later.
                            </div>
                            <div
                              class="my-6 p-6 bg-gray-200"
                            >
                              <div
                                class="font-semibold font-alt-sans text-lg pb-2"
                              >
                                Important
                              </div>
                              <div>
                                Read and follow all application instructions carefully.
                              </div>
                            </div>
                            <a
                              class="button is-info is-small"
                              href="https://www.sf.gov/sites/default/files/2024-04/BMR Homeownership FCFS Full Application 04.2024.pdf"
                              target="_blank"
                            >
                              Download PDF application
                            </a>
                          </div>
                        </li>
                        <li>
                          <h3
                            class="text-heading-xl font-semibold font-alt-sans pb-4"
                          >
                            Gather documents
                          </h3>
                          <div
                            class="text-base"
                          >
                            <div
                              class="text-base"
                            >
                              You must provide documents for everyone you will be living with who is over 18.
                            </div>
                            <div
                              class="my-6 p-6 bg-gray-200"
                            >
                              <div
                                class="font-semibold font-alt-sans text-lg pb-2"
                              >
                                What documents you need.
                              </div>
                              <div>
                                See first 2 pages of the PDF application to find out what documents you need.
                              </div>
                            </div>
                          </div>
                        </li>
                        <li>
                          <h3
                            class="text-heading-xl font-semibold font-alt-sans pb-4"
                          >
                            Combine everything into 1 PDF file
                          </h3>
                          <div
                            class="text-base"
                          >
                            <div
                              class="text-base"
                            >
                              Save your documents and filled out application into 1 PDF file.
                            </div>
                            <div
                              class="my-6 p-6 bg-gray-200"
                            >
                              <div
                                class="font-semibold font-alt-sans text-lg pb-2"
                              >
                                Name your file like this
                              </div>
                              <div>
                                <div>
                                  Name of building - Last name, First name
                                </div>
                                <div
                                  class="pt-4 italic"
                                >
                                  <div>
                                    Example
                                  </div>
                                  <div>
                                    <span
                                      class="ui-icon ui-medium"
                                    >
                                      <svg
                                        fill="currentColor"
                                        viewBox="0 0 38 51"
                                      >
                                        <path
                                          d="M36.1495 12.6898L25.6104 2.28944C24.413 1.1119 22.8099 0.453501 21.1276 0.453501L6.33333 0.453491C2.83516 0.453491 0 3.25134 0 6.70349L0.000643229 44.2035C0.000643229 47.6556 2.8358 50.4535 6.33398 50.4535H31.6667C35.1648 50.4535 38 47.6556 38 44.2035V17.1039C38 15.4437 37.337 13.8617 36.1495 12.6898ZM22.1667 3.78163C22.6097 3.93456 23.0242 4.15721 23.366 4.49442L33.9051 14.8948C34.2495 15.2289 34.4771 15.639 34.6255 16.0785H23.75C22.8792 16.0785 22.1667 15.3754 22.1667 14.516V3.78163ZM34.8333 44.2035C34.8333 45.9262 33.4123 47.3285 31.6667 47.3285H6.33333C4.58771 47.3285 3.16667 45.9262 3.16667 44.2035V6.7035C3.16667 4.98084 4.58771 3.5785 6.33333 3.5785H19V14.516C19 17.1039 21.1276 19.2035 23.75 19.2035H34.8333V44.2035ZM9.5 27.016C9.5 27.8754 10.2125 28.5785 11.0833 28.5785H26.9167C27.7875 28.5785 28.5 27.8754 28.5 27.016C28.5 26.1566 27.7875 25.4535 26.9167 25.4535H11.0833C10.2125 25.4535 9.5 26.1566 9.5 27.016ZM26.9167 31.7035H11.0833C10.2125 31.7035 9.5 32.4066 9.5 33.266C9.5 34.1254 10.2125 34.8285 11.0833 34.8285H26.9167C27.7913 34.8285 28.5 34.1289 28.5 33.266C28.5 32.4031 27.7875 31.7035 26.9167 31.7035ZM26.9167 37.9535H11.0833C10.2125 37.9535 9.5 38.6566 9.5 39.516C9.5 40.3754 10.2125 41.0785 11.0833 41.0785H26.9167C27.7913 41.0785 28.5 40.3789 28.5 39.516C28.5 38.6531 27.7875 37.9535 26.9167 37.9535Z"
                                        />
                                      </svg>
                                    </span>
                                    1288 Howard - Chow, Tania
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              class="text-base"
                            >
                              <span>
                                If you cannot combine everything into 1 file, you can upload multiple files. Name each file as shown above and include the type of document. 
                                <i>
                                  Example: 1288 Howard - Chow, Tania - Proof of Income
                                </i>
                              </span>
                            </div>
                          </div>
                        </li>
                        <li>
                          <h3
                            class="text-heading-xl font-semibold font-alt-sans pb-4"
                          >
                            Create a free account at Box.com
                          </h3>
                          <div
                            class="text-base"
                          >
                            <div
                              class="text-base pb-2"
                            >
                              <span>
                                <a
                                  class="underline"
                                  href="https://account.box.com/signup/personal?tc=annual"
                                  target="_blank"
                                >
                                  Create a Box.com account.
                                </a>
                                 In the next step, you will need a free Box.com account to submit your application.
                              </span>
                            </div>
                            <div
                              class="text-base"
                            >
                              If you already have an account, use that one.
                            </div>
                            <div
                              class="how-to-apply-dropdown pt-2"
                            >
                              <button
                                aria-expanded="false"
                                class="button is-unstyled m-0 no-underline has-toggle"
                                type="button"
                              >
                                What is Box.com
                              </button>
                            </div>
                          </div>
                        </li>
                        <li>
                          <h3
                            class="text-heading-xl font-semibold font-alt-sans pb-4"
                          >
                            Submit your application and get your place in line
                          </h3>
                          <div
                            class="text-base"
                          >
                            <div
                              class="text-base"
                              id="SubmitApplicationStep"
                            >
                              When you are ready, submit your application in 3 steps:
                            </div>
                            <ul
                              class="mb-0 pb-2"
                            >
                              <li
                                class="text-base"
                              >
                                <span>
                                  Fill out your information and get a link to upload your 
                                  <a
                                    class="underline"
                                    href="#HowToApplySection"
                                  >
                                    PDF application and documents
                                  </a>
                                </span>
                              </li>
                              <li
                                class="text-base"
                              >
                                Follow the link to Box.com
                              </li>
                              <li
                                class="text-base"
                              >
                                Upload your filled out application and all documents to Box.com
                              </li>
                            </ul>
                            <div
                              class="text-base"
                            >
                              The time and date you apply determines your place in line.
                            </div>
                            <div
                              class="text-base pt-6"
                            >
                              <span
                                class="ui-icon ui-medium"
                              >
                                <svg
                                  fill="currentColor"
                                  viewBox="0 0 32 32"
                                >
                                  <path
                                    d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                                  />
                                  <path
                                    d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                                  />
                                </svg>
                              </span>
                               Applications open January 1, 2050 at 1:00 AM Pacific Time. Check back here to apply.
                            </div>
                            <div
                              class="my-6 p-6 bg-gray-200"
                            >
                              <div
                                class="font-semibold font-alt-sans text-lg pb-2"
                              >
                                Upload your application within 30 minutes
                              </div>
                              <div>
                                If you do not upload your application and all documents within 30 minutes, you will lose your place in line. You will get a new place in line based on the time and date we receive your completed application.
                              </div>
                            </div>
                          </div>
                        </li>
                      </ol>
                    </div>
                    <h2
                      class="text-2xl font-alt-serif"
                    >
                      What happens next
                    </h2>
                    <div
                      class="pt-4"
                    >
                      <span>
                        We will email you your place in line within 
                        <b>
                          2 business days
                        </b>
                        . If anyone applied before you, we review their application(s) first.
                      </span>
                    </div>
                    <div
                      class="py-2"
                    >
                      <span>
                        If your application is approved, you will need to enter a purchase agreement within 
                        <b>
                          7 days
                        </b>
                        .
                      </span>
                    </div>
                    <a
                      aria-label="Learn more about how it works to apply"
                      class="underline"
                      href="https://www.sf.gov/node/14246"
                      target="_blank"
                    >
                      Learn more about how it works
                    </a>
                  </div>
                </div>
              </article>
            </section>
          </div>
          <section
            class="aside-block w-full md:w-1/3 md:max-w-xs text-gray-950 md:border-l border-t md:border-t-0 border-gray-450 border-b-0 p-6 mx-0"
          >
            <h2
              class="text__underline-weighted"
            >
              Need Help?
            </h2>
            <div
              class="text-sm text-gray-750"
            >
              <p
                class="mb-4"
              >
                <span>
                  Housing counselors can help with your application. Support is available in many languages.
                </span>
              </p>
              <a
                class="button w-full transition"
                href="https://www.sf.gov/resource/2022/homebuyer-program-counseling-agencies"
                target="_blank"
              >
                Find a Housing Counselor
              </a>
            </div>
          </section>
        </article>
      </main>
    </div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section pb-8"
        >
          <img
            alt=""
            data-testid="footer-logo-test-id"
          />
        </div>
      </div>
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section py-2"
        >
          <p
            class="text-gray-500"
          >
            <span>
              DAHLIA: San Francisco Housing Portal is a project of the 
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/mayors-office-housing-and-community-development"
                target="_blank"
              >
                Mayor's Office of Housing and Community Development
              </a>
            </span>
          </p>
          <p
            class="text-xs mt-4 text-gray-500"
          >
            <span>
              in partnership with
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/city-administrator/digital-services"
                target="_blank"
              >
                San Francisco Digital Services
              </a>
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/mayors-office-innovation"
                target="_blank"
              >
                Mayor's Office of Civic Innovation
              </a>
            </span>
          </p>
        </div>
      </div>
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section pb-8"
        >
          <p
            class="text-sm"
          >
            For listing and application questions, please contact the agent displayed on the listing. 
            <br />
            For general program inquiries, you may call MOHCD at 415-701-5500.
          </p>
        </div>
      </div>
      <section
        class="footer-sock"
      >
        <div
          class="footer-sock__inner"
        >
          <p
            class="footer-copyright"
          >
            © City & County of San Francisco
          </p>
          <nav
            aria-label="Footer"
            class="footer-nav"
          >
            <a
              class="text-gray-500"
              href="https://airtable.com/shrw64DubWTQfRkdo"
              target="_blank"
            >
              Give Feedback
            </a>
            <a
              class="text-gray-500"
              href="mailto:sfhousinginfo@sfgov.org"
              target="_self"
            >
              Contact
            </a>
            <a
              class="text-gray-500"
              href="/disclaimer"
              target="_self"
            >
              Disclaimer
            </a>
            <a
              class="text-gray-500"
              href="/privacy"
              target="_self"
            >
              Privacy Policy
            </a>
          </nav>
        </div>
      </section>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<HowToApply /> shows the correct leasing agent box for the listing 1`] = `
<DocumentFragment>
  <div
    class="leasing-agent"
  >
    <p
      class="m-0"
    >
      Matthew Burns
    </p>
    <p
      class="text-gray-750 text-sm"
    >
      Leasing Agent
    </p>
    <a
      href="tel:415 5552323"
    >
      <span
        class="ui-icon ui-medium pr-2"
      >
        <svg
          fill="#0077DA"
          viewBox="0 0 32 32"
        >
          <path
            d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
          />
        </svg>
      </span>
      Call (415) 555-2323
    </a>
    <div
      class="pb-3 pt-1"
    >
      <a
        href="mailto:ae@email.com"
      >
        <span
          class="pr-2"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope "
            data-icon="envelope"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
              fill="currentColor"
            />
          </svg>
        </span>
        Email
      </a>
    </div>
    <h3
      class="text-heading-sm"
    >
      Office hours
    </h3>
    <p
      class="text-sm"
    >
      8-4 Mon-Thurs
9-4 Fri
10-2 Weekends
    </p>
  </div>
</DocumentFragment>
`;
